processing/src/main/java/io/druid/query/lookup/LookupSnapshotTaker.java

```diff
@@ -62,11 +60,12 @@ public LookupSnapshotTaker(
     if (!this.persistDirectory.isDirectory()) {
       throw new ISE("Can only persist to directories, [%s] wasn't a directory", persistDirectory);
     }
-    this.persistFile = new File(persistDirectory, PERSIST_FILE_NAME);
   }
 
-  public synchronized List<LookupBean> pullExistingSnapshot()
+  public synchronized List<LookupBean> pullExistingSnapshot(final String tier)
   {
+    final File persistFile = getPersistFile(tier);
```

Comment:
for backward compatibility...
if tier.lookupSnapshot.json is not found then we should look for lookupSnapshot.json (move that to tier.lookupSnapshot.json if exists).

let us leave some comment telling why that code exists can be removed some releases after 0.13.0 .

Reply:
I was thinking backwards compatible code is not really necessary, since this feature will be able to pull lookups from the coordinator if the file is not found.

Hopefully, this is fine during a rolling upgrade, where the user is paying attention and is sure the coordinator is online and everything is healthy. Meaning snapshots aren't really needed.

Reply:
What do you think?

Reply:
yes, if coord is up then snapshot doesn't matter. so its not very critical to be backward compatible in this case.